clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -928,39 +927,94 @@ public String toRst() {
         StringBuilder b = new StringBuilder();
 
         for (ConfigKey def : configs) {
-            b.append("``");
-            b.append(def.name);
-            b.append("``\n");
-            for (String docLine : def.documentation.split("\n")) {
-                if (docLine.length() == 0) {
-                    continue;
+            b.append(getConfigKeyRst(def, b));
```

Comment:
Since now the `StringBuilder` is being appended to directly, the `b.append` here would cause duplication of the accumulated content I think